### PR TITLE
Adapt for submodules

### PR DESCRIPTION
**Pull request by MykolaSuperman**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[submodule "external/aos_core_common_cpp"]
```
Comment (al1img) [inline .gitmodules:1]: Is it added manually by git command? If manually, add empty line between items.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+	path = external/aos_core_common_cpp
+	url = https://github.com/MykolaSolyanko/aos_core_common_cpp.git
+[submodule "external/api/aos_core_api"]
```
Comment (al1img) [inline .gitmodules:4]: Shall aos core lib be added at same level here?

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[submodule "external/aos_core_common_cpp"]
+	path = external/aos_core_common_cpp
+	url = https://github.com/MykolaSolyanko/aos_core_common_cpp.git
```
Comment (al1img) [inline .gitmodules]: Readme info how to build with submodules should be added?

File: .gitmodules (hunk 1 above)
Comment (MykolaSuperman) [reply .gitmodules:1]: no, it's automatically `git` command

File: .gitmodules (hunk 2 above)
Comment (MykolaSuperman) [reply .gitmodules:4]: aos core lib is a submodule of common lib. So common lib includes core lib as a submodule

File: .gitmodules (hunk 3 above)
Comment (MykolaSuperman) [reply .gitmodules]: Done

File: CMakeLists.txt [hunk 4]
```diff
@@ -122,9 +120,14 @@ endif()
 # Targets
 # ######################################################################################################################
 
+add_subdirectory(external/aos_core_common_cpp/external/aos_core_lib_cpp/src/common)
+add_subdirectory(external/aos_core_common_cpp/external/aos_core_lib_cpp/src/iam)
+add_subdirectory(external/api)
+add_subdirectory(external/aos_core_common_cpp/src)
```
Comment (al1img) [inline CMakeLists.txt:126]: There is no cmake file in src. Is there?

Comment (MykolaSuperman) [reply CMakeLists.txt:126]: The cmake file is indeed present in the src folder for the common library.

**Review by mykola-kobets-epam — APPROVED** [review]

File: tests/communication/communication_secure_test.cpp [hunk 5]
```diff
@@ -201,11 +201,11 @@ class CommunicationSecureManagerTest : public ::testing::Test {
         // add CA certificate to the chain
         aos::StaticString<aos::crypto::cCertPEMLen> caCert;
 
-        ASSERT_TRUE(aos::FS::ReadFileToString(CERTIFICATES_DIR "/ca.cer", caCert).IsNone());
+        ASSERT_TRUE(aos::FS::ReadFileToString(CERTIFICATES_MP_DIR "/ca.cer", caCert).IsNone());
         clientCertChain.Append(caCert);
 
         // apply client certificate
-        // FS::WriteStringToFile(CERTIFICATES_DIR "/client-out.pem", clientCertChain, 0666);
+        // FS::WriteStringToFile(CERTIFICATES_MP_DIR "/client-out.pem", clientCertChain, 0666);
```
Comment (al1img) [inline tests/communication/communication_secure_test.cpp]: Do you need these comments?

File: tests/communication/communication_secure_test.cpp [hunk 6]
```diff
@@ -230,8 +230,8 @@ class CommunicationSecureManagerTest : public ::testing::Test {
             ENGINE_get_finish_function(engine)(engine);
         }
 
-        // aos::FS::ClearDir(SOFTHSM_BASE_DIR "/tokens");
-        std::filesystem::remove_all(SOFTHSM_BASE_DIR "/tokens");
+        // aos::FS::ClearDir(SOFTHSM_BASE_MP_DIR "/tokens");
```
Comment (al1img) [inline tests/communication/communication_secure_test.cpp]: ditto

Comment (MykolaSuperman) [reply tests/communication/communication_secure_test.cpp]: Done

File: tests/communication/communication_secure_test.cpp (hunk 5 above)
Comment (MykolaSuperman) [reply tests/communication/communication_secure_test.cpp]: Done

**Review by al1img — APPROVED** [review]

**Pull request merged**